openforcefield/tests/test_forcefield.py

```diff
@@ -1112,6 +1179,161 @@ def test_freesolv_parameters_assignment(self, freesolv_id, forcefield_version, a
+
+        # Use GB params from OpenMM GBSA classes to populate parameters
+        for idx, (radius, screen) in enumerate(gb_params):
+            q, _, _2 = amber_gbsa_force.getParticleParameters(idx)
+            print(q, radius, screen, _, _2)
+            if isinstance(amber_gbsa_force, openmm.GBSAOBCForce):
+                # Note that in GBSAOBCForce, the per-particle parameters are separate
+                # arguments, while in CustomGBForce they're a single iterable
+                amber_gbsa_force.setParticleParameters(idx, q, radius, screen)
+
+            elif isinstance(amber_gbsa_force, openmm.CustomGBForce):
+                # !!! WARNING: CustomAmberGBForceBase expects different per-particle parameters
+                # depending on whether you use addParticle or setParticleParameters. In
+                # setParticleParameters, we have to apply the offset and scale BEFORE setting
+                # parameters, whereas in addParticle, it is applied afterwards, and the particle
+                # parameters are not set until an auxillary finalize() method is called. !!!
+                amber_gbsa_force.setParticleParameters(idx, (q, radius - 0.009, screen * (radius - 0.009)))
+
+
+        # Put the GBSA force into a separate group so we can specifically compare GBSA energies
+        amber_gbsa_force.setForceGroup(1)
+        off_gbsa_force.setForceGroup(1)
+
+        # Some manual overrides to get the OFF system's NonbondedForce matched up with the AMBER system
+        if is_periodic:
+            off_nonbonded_force.setNonbondedMethod(openmm.NonbondedForce.CutoffPeriodic)
+        else:
+            off_nonbonded_force.setNonbondedMethod(openmm.NonbondedForce.NoCutoff)
+
+        off_nonbonded_force.setReactionFieldDielectric(1.0)
+
+        # Not sure if zeroing the switching width is essential -- This might only make a difference
+        # in the energy if we tested on a molecule larger than the 9A cutoff
+        #off_nonbonded_force.setSwitchingDistance(0)
+
+        # Create Contexts
+        integrator = openmm.VerletIntegrator(1.0 * unit.femtoseconds)
+        amber_context = openmm.Context(amber_omm_system, integrator)
+        off_context = openmm.Context(off_omm_system, copy.deepcopy(integrator))
+
+        # Get context energies
+        amber_energy = get_context_potential_energy(amber_context, positions)
+        off_energy = get_context_potential_energy(off_context, positions)
+
+        # Very handy for debugging
+        # print(openmm.XmlSerializer.serialize(off_gbsa_force))
+        # print(openmm.XmlSerializer.serialize(amber_gbsa_force))
+
+        # Ensure that the GBSA energies (which we put into ForceGroup 1) are identical
+        assert amber_energy[1] == off_energy[1]
+
+        # Ensure that all system energies are the same
+        compare_system_energies(off_omm_system, amber_omm_system, positions, by_force_type=False)
+
```

Comment:
Do you actually test that FreeSolve free energies are equivalent?

Reply:
No -- Just single conformer energies.